esp-hal/src/aes/mod.rs

```diff
@@ -427,7 +496,7 @@ pub mod dma {
             read_buffer_len: usize,
             mode: Mode,
             cipher_mode: CipherMode,
-            key: [u8; 16],
+            key: Key,
         ) -> Result<(), crate::dma::DmaError> {
             // AES has to be restarted after each calculation
             self.reset_aes();
```

Comment:
One last thing sorry, and I know this isn't code you touched but could you make `reset_aes`, `set_mode` and `set_cipher_mode` on `AesDma` not `pub`. It's only used internally, and it can break our abstraction being public

Reply:
These last four functions: https://docs.esp-rs.org/esp-hal/esp-hal/0.16.1/esp32c6/esp_hal/aes/dma/struct.AesDma.html#method.reset_aes

Reply:
Done, no problem!